src/udev/udev-watch.c

```diff
@@ -90,12 +91,22 @@ int udev_watch_begin(int inotify_fd, sd_device *dev) {
         r = mkdir_parents(filename, 0755);
         if (r < 0)
                 return log_device_error_errno(dev, r, "Failed to create parent directory of '%s': %m", filename);
-        (void) unlink(filename);
 
         r = device_get_id_filename(dev, &id);
         if (r < 0)
                 return log_device_error_errno(dev, r, "Failed to get device id-filename: %m");
 
+        r = readlink_malloc(filename, &old_id);
+        if (r >= 0) {
+                if (streq(id, old_id))
```

Comment:
if I understand correctly these are paths, so path_equal() ?

Reply:
`path_equal()` handles slashes gracefully. But in this case, the device id must be unique, hence should be handled strictly. Moreover, in most cased, the ids saved in symlinks do not look like paths:
```
$ ls -l /run/udev/watch
total 0
lrwxrwxrwx 1 root root 4 Mar  5 08:50 1 -> b8:0
lrwxrwxrwx 1 root root 6 Mar  5 08:49 2 -> b259:2
lrwxrwxrwx 1 root root 6 Mar  5 08:49 3 -> b259:1
lrwxrwxrwx 1 root root 6 Mar  5 08:49 4 -> b259:0
lrwxrwxrwx 1 root root 4 Mar  5 18:41 666 -> b7:0
lrwxrwxrwx 1 root root 4 Mar  5 10:43 70 -> b7:1
```

Reply:
I see, makes sense